prov/rxm/src/rxm_conn.c

```diff
@@ -1379,6 +1412,92 @@ static void *rxm_conn_progress(void *arg)
 	return NULL;
 }
 
+static void *rxm_conn_atomic_progress(void *arg)
+{
+	struct rxm_ep *rxm_ep = container_of(arg, struct rxm_ep, util_ep);
+	struct rxm_fabric *rxm_fabric;
+	struct fid *fids[2];
+	struct pollfd fds[2];
+	struct rxm_msg_eq_entry *entry;
+	int poll_required;
+	int nfids;
+	int ret;
+
+	assert(rxm_ep->msg_eq);
+	rxm_fabric = container_of(rxm_ep->util_ep.domain->fabric,
+				  struct rxm_fabric, util_fabric);
+	entry = calloc(1, RXM_MSG_EQ_ENTRY_SZ);
+	if (!entry) {
+		FI_WARN(&rxm_prov, FI_LOG_EP_CTRL,
+			"Unable to allocate memory!\n");
+		return NULL;
+	}
+
+	fids[0] = &rxm_ep->msg_eq->fid;
+	ret = fi_control(fids[0], FI_GETWAIT, (void *) &fds[0].fd);
+	if (ret) {
+		FI_WARN(&rxm_prov, FI_LOG_EP_CTRL,
+			"Unable to get MSG_EP EQ WAIT_FD %d\n", ret);
+		goto exit;
+	}
+	fds[0].events = POLLIN;
+	nfids = 1;
+
+	FI_DBG(&rxm_prov, FI_LOG_EP_CTRL,
+	       "Starting CM conn thread with atomic AUTO_PROGRESS\n");
+
+	while (1) {
+		ofi_ep_lock_acquire(&rxm_ep->util_ep);
+		/* CQ can be bound after progress thread is started */
+		if (OFI_UNLIKELY(nfids == 1 && rxm_ep->msg_cq)) {
+			nfids = 2;
+			fids[1] = &rxm_ep->msg_cq->fid;
+			fds[1].fd = rxm_ep->msg_cq_fd;
+			fds[1].events = POLLIN;
+		}
+		poll_required = fi_trywait(rxm_fabric->msg_fabric, fids, nfids);
```

Comment:
Please change it to `!fi_trywait(...)`, so that we do poll when `poll_required` is True.

Reply:
That was a bad variable name choice on my part, I'll change it so that it is more clear.